### DISCUSSION TITLE
response speed is too slow

### DISCUSSION THREAD

**Discussion by godoge** [open-webui/open-webui#7821] [General]
I chat with other tools, and the replies are not this slow; Open-WebUI's responses are too slow.

**Comment by 0xGitGuy** [comment]
Having the same issue, Open Web UI seems to be about 10 seconds slower in return the full response than native playground at OpenAI GPT-o1 Mini.  Time to first chunk is also a few seconds slower on average. OpenWeb UIs is constantly be polling in browser network inspection, but the native playground does not. I'm not sure if this is the best way to handle the completions.

**Comment by ghost** [comment]
Try switching off the 'Retrieval Query Generation', 'Tags Generation' and 'Autocomplete Generation' (all of them may depend a lot on the HW you are running this on) in: Admin Panel - Settings - Interface, if it's not a problem for you to don't have that readily available.

Restart open-webui (not just refresh web UI, **restart** entire process), maybe Ollama as well if GPU still have VRAM occupied.

**Reply by triggerman722** [reply]
100% fixed my case. Speed is night and day better.

**Reply by liaohuanxuan** [reply]
Version 0.9.6 is still useful.

**Comment by 0xGitGuy** [comment]
Where do you find autocomplete generation?  Can't find it in the settings > interface for me. I offed Retrieval query generation, doesn't help.

**Reply by jsun1973** [reply]
In the top right of the Open-WebUI, click the picture associated with your account.  Go to settings, then click admin settings. From there, select interface on the left.

**Reply by 0xGitGuy** [reply]
I think autocomplete generation meant, " default prompt suggestions".  I turned all three off and doesn't seem to make any difference.  Restarted ollama and open webui, nothing helps. 

Comparing OpenAI chat playground and dockerized open webui setup, do you experience same speed of response after prompting or open webui is always much slower?

**Reply by jsun1973** [reply]
I just tested the cli version of ollama without the webUI and it is nice and speedy.  I would say there is a slight difference using WebUI - it's a bit slower.  But overall when I turned off those settings, it did speed it up significantly.  I know depending on the LLM library, some are more efficient than others so you might want to try a different one.  I am using llama 3.2.

Keep in mind I am brand new to this, so I'm not well versed.

**Reply by 0xGitGuy** [reply]
using the webui, if i select openai o1-mini model, does it even use llama 3.2 at all? I mainly use it with external model provider for speed reasons so i can work faster without a beefy setup

**Reply by 0xGitGuy** [reply]
Settings>Interface>Set Task Model.  you can specify under the task model, local models/external models. 

didnt make a difference even choosing the fastest model for task. I wonder if it's an issue of using 1vpu in ubuntu.

**Comment by clduab11** [comment]
Just as an FYI, what is going on (at least for my end) is listed here...

[https://github.com/ollama/ollama/issues/7919#issuecomment-2560395085](url)

If you use AutoComplete Generation, and your context window is set _outside_ of the 2048 window (Ollama default), the llama_runner mechanism evicts the current model and reloads it for the AutoComplete Generation...then it has to do that function, and once the text generates, THEN you have to kick THAT model off to reload the model with your settings (like num_ctx) to actually _prompt_ your model...rinse/repeat. It severely degrades the time to first token, as well as further conversational follow-up.

I ended up setting my num_threads and num_ctx at the model level (instead of the system level, kind of a pain, but eh...) and disabling my AutoComplete Generation and my local models vastly improved. Rick says there is a PR on the Ollama side that's meant to fix this issue (which is linked here... [https://github.com/ollama/ollama/pull/8029](url)), but it isn't yet ready for integration, so I'm assuming this is a temporary solution and it'll be fixed in a downstream version.

**Reply by 0xGitGuy** [reply]
Would this matter if I exclusively use external models like OpenAI's 01mini?

**Reply by clduab11** [reply]
>  Would this matter if I exclusively use external models like OpenAI's 01mini? 

Honestly, I wouldn't think so, as I use external models via API as well...and those have never been an issue. 

But I suppose that, given the API might be called for the AutoGenerate, there might be some slowdown depending on policies for multiple requests? I saw in the other thread your post re: 1vpu for Ubuntu, it very well could be that, given you tried turning all options off including AutoComplete? 

But I'm still pretty new to all this, and that's really just a guess 😅. My configuration is running OWUI/Ollama in Windows via Docker (I use Docker Compose and launch with a .yaml), so I can't speak to Ubuntu unfortunately.

**Comment by 0xGitGuy** [comment]
I finally found out the issue.  It's not because of the hardware resources. It's because of the implementation of SSL certificate.  Once the SSL is turned off, the GPT-4o mini responds very fast. But using HTTPS should not slow down the response from external APIs.  I'm guessing streaming the external API output chunks under the SSL slows things down.  Are there any settings that I can change to speed this up?

**Reply by harrywang** [reply]
I am also running into this issue, running fast locally, but once deployed, super super slow. don't know why, have not tried turning off ssl, which should not be down in production anyway.

**Reply by harrywang** [reply]
@0xGitGuy have you figured out any solution yet?

**Reply by harrywang** [reply]
seems related to https://github.com/open-webui/open-webui/issues/8374

**Reply by harrywang** [reply]
i don't think our problem is about ssl, it's postgres database issue

**Reply by 0xGitGuy** [reply]
> @0xGitGuy have you figured out any solution yet?

 I can't remember how I fixed it, but now I'm on SSL and it's working fine. My best guess, it had to do with the Nginx server settings in the default.conf file

**Comment by harrywang** [comment]
https://github.com/user-attachments/assets/e19f40cf-9ad6-4fae-83a5-0dc460c1ead4

this is how slow it gets

**Reply by harrywang** [reply]
change the connection pool size to 30 made it much faster but still slower than sqlite: https://github.com/open-webui/open-webui/discussions/8665

**Comment by limdingwen** [comment]
I found my issue. Getting the user location takes a long time (10 seconds) on Firefox Linux. This is the offending code (in `Chat.svelte`):

```
		let userLocation;
		if ($settings?.userLocation) {
			userLocation = await getAndUpdateUserLocation(localStorage.token).catch((err) => {
				console.error(err);
				return undefined;
			});
		}
```

I don't need the user location anyway. To fix this, either block the website's access to user location, or disable it in your OWUI settings.
